### DISCUSSION TITLE
allow per-file skip_list or whole-file `noqa`

### DISCUSSION THREAD

**Discussion by eengstrom** [ansible/ansible-lint#2843] [Ideas]
In a couple of places I've wanted to be able to exclude some lint rules from applying to an entire file.  
e.g. `name[play]` where the entire playbook consists of nothing but `- import_playbook: some-playbook.yml`.
I have looked at the documentation and found no such feature.

rudimentary ideas I've come up with so far:

## allowing `skip_list[_by_path]` 

e.g. in `.ansbile-lint`

```
skip_list_by_path:
  aggregator.yml:
    - name[play]
```

## in-file `noqa[-file]` block

e.g. at or near the top of a file:

```
# Ignore these ansible-lint rules for the entire file:
# noqa: name[play]
# or 
# noqa-file: name[play]
```

**Comment by dato** [comment]
This can be achieved with: [docs/lint/configuring/#ignoring-rules-for-entire-files](https://ansible.readthedocs.io/projects/lint/configuring/#ignoring-rules-for-entire-files)

(but note https://github.com/ansible/ansible-lint/issues/3068).

**Reply by eengstrom** [reply]
While I agree that what you suggest **partially** solves the problem I outline, I have several issues with that approach:

First, if used as a `.ansible-lint-ignore`, as opposed to a `.config/ansible-lint-ignore.txt`, it only works on filenames **in the current directory**, but not path names.  e.g.:

`playbook.yml name[play]`
  
works ... but this does not work:

`path/to/playbook.yml name[play]`

That means you need to have directory specific versions, unlike say `.gitignore`.  Plausible workaround - use symlinks to a single copy, I guess.

Second, when the Ansible Lint docs say "ignore", the lint rules defined like those don't really "ignore" that check.  Instead, it is marked as a WARNING, not an ERROR. The upshot of that is that the ignore-d rule is still in the output, and then downstream tools, such as VSCode, still pick up on it and present it as a problem.  That's not really "ignored".  This is, as you mentioned, already discussed here:

https://github.com/ansible/ansible-lint/issues/3068

Third, this feature require **each** file to be enumerated.
 
IMO, it would be way nicer if `ansible-lint` followed the `yamllint` config idea:

https://yamllint.readthedocs.io/en/latest/configuration.html#ignoring-paths
